src/main/java/com/marketdata/MarketSimulatorBackend/service/AlphaVantageProvider.java

```diff
@@ -0,0 +1,49 @@
+package com.marketdata.MarketSimulatorBackend.service;
+
+import com.marketdata.MarketSimulatorBackend.model.StockPrice;
+import org.json.JSONObject;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class AlphaVantageProvider implements StockServiceProvider {
+
+    private static final String ALPHA_VANTAGE_URL = "https://www.alphavantage.co/query";
+    private static final String API_KEY = "OYPLC5RZ9VIXXVG5"; // Replace with your actual key
```

Comment:
yes, this should not be hardcoded in the code, should be used from .env file for example, and don't push the env file, you can apply it later, it's okay like this for now.